### ISSUE TITLE
Bump Spark33 version to Spark3.3.3

### ISSUE THREAD

**Issue by zwangsheng** [apache/incubator-gluten#4896]
### Description

As title

**Comment by PHILO-HE** [comment]
@weiting-chen, do we have any users using Spark 3.3.1? If no, directly upgrading to 3.3.4 may be good. cc @zhouyuan.

**Comment by zhouyuan** [comment]
@zwangsheng @PHILO-HE  do you see if we could support all minor releases for Spark 3.3?

thanks,
-yuan

**Comment by PHILO-HE** [comment]
Hi @zwangsheng, did you find any issue in your spark-3.3.4 env. by directly using Gluten jar built for spark-3.3.1? I think there should have some incompatible issues for parquet write.

**Comment by zwangsheng** [comment]
> Hi @zwangsheng, did you find any issue in your spark-3.3.4 env. by directly using Gluten jar built for spark-3.3.1? I think there should have some incompatible issues for parquet write.

Yeah, when try to run the exists unit tests, I found Gluten had built-in `ParquetFileFormat.scala`, which has been changed in Spark3.3.4. This may cause parquet file read or write issue.

**Comment by PHILO-HE** [comment]
> > Hi @zwangsheng, did you find any issue in your spark-3.3.4 env. by directly using Gluten jar built for spark-3.3.1? I think there should have some incompatible issues for parquet write.
> 
> Yeah, when try to run the exists unit tests, I found Gluten had built-in `ParquetFileFormat.scala`, which has been changed in Spark3.3.4. This may cause parquet file read or write issue.
> 
> You can found the diff with the draft PR #4897

`ParquetFileFormat.scala` was ported from Spark code with a few code changes that overwrite Spark's logic. Maybe, we have to maintain two such files respectively for 3.3.1 & 3.3.4 in shims module to support the both versions.
Can we just keep using 3.3.1 UTs in Gluten? Assume most UTs are shared by 3.3.1 & 3.3.4. Maybe, we can just extract some important UTs ( not covered in 3.3.1) from spark-3.3.4 and put into a new test module in gluten.

**Comment by zwangsheng** [comment]
> `ParquetFileFormat.scala` was ported from Spark code with a few code changes that overwrite Spark's logic. Maybe, we have to maintain two such files respectively for 3.3.1 & 3.3.4 in shims module to support the both versions. Can we just keep using 3.3.1 UTs in Gluten? Assume most UTs are shared by 3.3.1 & 3.3.4. Maybe, we can just extract some important UTs ( not covered in 3.3.1) from spark-3.3.4 and put into a new test module in gluten.

I see. I will give a try to maintain two files. And go deep to find the difference.
